frontend/src/metabase/visualizations/echarts/cartesian/model/dataset.ts

```diff
@@ -577,3 +577,24 @@ export const getBubbleSizeDomain = (
 
   return [0, bubbleSizeDomainMax];
 };
+
+export function getHistogramDataset(
+  dataset: ChartDataset,
+  xAxisModel: XAxisModel,
+) {
+  if (!isCategoryAxis(xAxisModel) || !xAxisModel.isHistogram) {
+    return dataset;
+  }
+
+  const interval = xAxisModel.histogramInterval ?? 1;
+
+  dataset.unshift({
```

Comment:
This should be breaking tooltips and click behavior since now the `dataIndex` of the transformed dataset does not match the dataIndex of the orignial dataset which stores required values

Reply:
<img width="1686" alt="Screenshot 2024-03-12 at 11 47 09 AM" src="https://github.com/metabase/metabase/assets/37751258/ba3fd753-50a1-4d71-beb6-692e95dad261">

Tooltips seem to work fine for me

Reply:
https://github.com/metabase/metabase/assets/37751258/a8c40821-def4-4e40-a723-3a86935b17b1

Click behavior as well

Reply:
I missed that initially you changed the raw dataset instead of the transformed one, so now there is an issue with tooltips and clicks since the dataIndex points to a wrong datum:
https://github.com/metabase/metabase/assets/14301985/f1f940c5-4429-4d09-9209-d2daeed4d344
But it should be straightforward to fix

Reply:
Ah that makes sense I'll fix it, I should just be able to offset the index if it's a histogram

Reply:
I think simply adding an offset won't work when there was `null` which was removed. But we can store the original datum index

Reply:
@alxnddr Should be fixed now